app/serializers.py

```diff
@@ -26,4 +26,24 @@ def create(self, validated_data):
             question = Question.objects.create(**question_data)
             poll.questions.add(question)
         
-        return poll
\ No newline at end of file
+        return poll
+    
+    def update(self, instance, validated_data):
+        questions_data = validated_data.pop('questions', None)
+        creater_email = validated_data.get('creater', instance.creater.email)
+
+        if creater_email != instance.creater.email:
+            user, created = User.objects.get_or_create(email=creater_email)
+            instance.creater = user
+
+        instance.title = validated_data.get('title', instance.title)
+        instance.is_open = validated_data.get('is_open', instance.is_open)
+        instance.save()
+
+        if questions_data is not None:
+     
+            for question_data in questions_data:
+                question = Question.objects.create(**question_data)
+                instance.questions.add(question)
+
+        return instance
```

Comment:
This is not good, it will add new questions as well as keep the old ones too...first delete the old questions & then set new..Also make sure title of question is unique..So change create & update methods to only create questions if doesn't exists already